models/semester.js

```diff
@@ -0,0 +1,52 @@
+import connector from "#models/databaseUtil";
+const semesterSchema = {
+  number:int , {enum: ["1", "2" , "3" , "4" , "5","6","7","8"], required:true},
+  academicYear: {
+    type: String,
+    required: true,
+    validate: {
+      validator: (value) => /^20\d{2}$/.test(value),
+      message: (props) => `${props.value} is not a valid year format starting with "2"!`,
+    },
+  },
+  type:{ enum: ["ODD", "EVEN"], required: true },
+
+  startDate: { type: Date, required: true },
+  endDate: { type: Date, required: true },
+};
+
+// eslint-disable-next-line  no-unused-vars
+const semester = connector.model("semester", semesterSchema);
+
+//  CURD operations
+async function create(semesterData) {
+    const {
+      number,
+      academicYear,
+    } = semesterData;
+    const semester = new semester({
```

Comment:
```suggestion
    const newSemester = new semester({
```
Make sure the name cannot be confused with another, and make sure that the same is done in the rest of the file.